llvm/lib/MC/MachObjectWriter.cpp

```diff
@@ -904,6 +908,18 @@ uint64_t MachObjectWriter::writeObject(MCAssembler &Asm) {
     unsigned Flags = Sec.getTypeAndAttributes();
     if (Sec.hasInstructions())
       Flags |= MachO::S_ATTR_SOME_INSTRUCTIONS;
+    if (!cast<MCSectionMachO>(Sec).isVirtualSection() &&
+        !isUInt<32>(SectionStart)) {
+      Asm.getContext().reportError(
+          SMLoc(), "Cannot encode offset of section; object file too large");
```

Comment:
`reportError` messages are generally not capitalized per https://llvm.org/docs/CodingStandards.html#error-and-warning-messages